dataprofiler/profilers/profile_builder.py

```diff
@@ -1857,6 +1864,52 @@ def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
 
         return corr_mat
 
+    def _update_chi2(self):
+        """
+        Calculates the p-value from a chi-squared test
+        for homogeneity between all categorical columns.
+        """
+        def _get_categorical_profile(col_idx):
+            """
+            Returns the CategoricalColumn associated with the given
+            column index. If the column is not determined to be categorical,
+            returns None.
+
+            :param col_idx: Column index
+            :type col_idx: int
+            """
+            compiler = self._profile[col_idx]
```

Comment:
I think this is actually the column profiler not the compiler which is below as `data_stats_compiler`. 

Do you think it would it make more sense to pass in the the column profiler into this function vs the index? Would the code below become more intuitive?